jib-core/src/main/java/com/google/cloud/tools/jib/filesystem/XdgDirectories.java

```diff
@@ -101,5 +106,59 @@ static Path getCacheHome(Properties properties, Map<String, String> environment)
     throw new IllegalStateException("Unknown OS: " + rawOsName);
   }
 
-  private UserCacheHome() {}
+  /**
+   * Returns {@code $XDG_CONFIG_HOME}, if available, or resolves the OS-specific user config home
+   * based.
+   *
+   * <p>For Linux, this is {@code $HOME/.config/}.
+   *
+   * <p>For Windows, this is {@code %LOCALAPPDATA%}.
+   *
+   * <p>For macOS, this is {@code $HOME/Library/Preferences/}.
+   */
+  @VisibleForTesting
+  static Path getConfigHome(Properties properties, Map<String, String> environment) {
+    // Use environment variable $XDG_CONFIG_HOME if set and not empty.
+    String xdgConfigHome = environment.get("XDG_CONFIG_HOME");
+    if (xdgConfigHome != null && !xdgConfigHome.trim().isEmpty()) {
+      return Paths.get(xdgConfigHome);
+    }
+
+    String userHome = properties.getProperty("user.home");
+    Path xdgPath = Paths.get(userHome, ".config");
+
+    String rawOsName = properties.getProperty("os.name");
+    String osName = rawOsName.toLowerCase(Locale.ENGLISH);
+
+    if (osName.contains("linux")) {
```

Comment:
We're now duplicating some of this platform-testing logic.  Is it worth extracting into separate `getOs(properties)` method that returns the platform type?  (I wish the JRE already provided this.)

Reply:
Might be a good idea if we ever need it more. But I think with how little logic it actually is, only duplicating in 2 places isn't so bad.